### PR TITLE
go.mod: replace the old GH urls with renamed ones.

### PR DESCRIPTION
**Pull request by sepehrdaddev**
Replaces the old GH urls that have been renamed with their new ones

### CODE REVIEW

**Comment by sepehrdaddev** [comment]
:+1:

File: go.mod [hunk 1]
```diff
@@ -182,3 +182,15 @@ require (
...
+
+replace github.com/docker/docker v24.0.7+incompatible => github.com/moby/moby v24.0.7+incompatible
+
+replace github.com/docker/distribution v2.8.2+incompatible => github.com/distribution/distribution v2.8.2+incompatible
+
+replace github.com/envoyproxy/protoc-gen-validate v1.0.2 => github.com/bufbuild/protoc-gen-validate v1.0.2
+
+replace github.com/uber/jaeger-lib v2.4.1+incompatible => github.com/jaegertracing/jaeger-lib v2.4.1+incompatible
+
+replace github.com/uber/jaeger-client-go v2.30.0+incompatible => github.com/jaegertracing/jaeger-client-go v2.30.0+incompatible
```
Comment (AlexanderYastrebov) [inline go.mod:196]: These I would just remove https://github.com/zalando/skipper/issues/2153

Comment (szuecs) [reply go.mod:196]: Not sure if we can just remove it for everyone @AlexanderYastrebov

**Comment by szuecs** [comment]
:-1:

**Comment by szuecs** [comment]
Replace directive breaks user code

**Pull request closed without merging**